src/accordion/styled.ts

```diff
@@ -0,0 +1,77 @@
+import {TextBlock} from '../text-block';
+import {
+  getResponsiveSize,
+  getResponsiveSpace,
+  getStylePreset,
+  getTypographyPreset,
+  styled,
+} from '../utils';
+import {logicalProps} from '../utils/logical-properties';
+import {AccordionProps} from './types';
+
+export const StyledAccodionContainer = styled.div``;
+
+export const StyledHeaderWrapper = styled(TextBlock)``;
+
+export const StyledAccordionButton = styled.button<
+  Pick<AccordionProps, 'overrides' | 'disabled'>
+>`
+  display: flex;
+  box-sizing: border-box;
+  align-items: center;
+  width: 100%;
+  cursor: ${({disabled}) => (disabled ? 'not-allowed' : 'default')};
+  & > :not(:last-of-type) {
+    ${getResponsiveSpace(
+      'marginRight',
+      'accordion.header',
+      'header',
+      'spaceInline',
+    )}
+  }
+  ${getResponsiveSize('minWidth', '', 'header', 'minWidth')};
+  ${getResponsiveSize('minHeight', '', 'header', 'minHeight')};
+  ${getStylePreset('accordion.header', 'header')};
```

Comment:
Don't you need to pass disabled prop to the getStylePreset to get styled of that?
```suggestion
  ${getStylePreset('accordion.header', 'header', {isDisabled: disabled})};
```